### PR TITLE
Derive type names analyzer from AbstractBuiltInCodeStyleDiagnosticAnalyzer

### LINKED ISSUE #62639: SimplifyTypeNamesDiagnosticAnalyzerBase should sub-type AbstractBuiltInCodeStyleDiagnosticAnalyzer
https://github.com/dotnet/roslyn/blob/8bd10f1548db958b1983f3bd236fcc56aafae1be/src/Features/Core/Portable/SimplifyTypeNames/SimplifyTypeNamesDiagnosticAnalyzerBase.cs#L31-L32

### PR DESCRIPTION
**Pull request by Youssef1313**
I tried to move the analyzer to Analyzers layer, but the language-specific implementations depend on internals of workspaces. But I kept the abstract base class in Analyzers.

Fixes #62639

### CODE REVIEW

File: src/Analyzers/Core/Analyzers/SimplifyTypeNames/SimplifyTypeNamesDiagnosticAnalyzerBase.cs [hunk 1]
```diff
@@ -72,25 +70,21 @@ internal abstract class SimplifyTypeNamesDiagnosticAnalyzerBase<TLanguageKindEnu
             helpLinkUri: DiagnosticHelper.GetHelpLinkForDiagnosticId(IDEDiagnosticIds.PreferBuiltInOrFrameworkTypeDiagnosticId),
             customTags: DiagnosticCustomTags.Unnecessary.Concat(EnforceOnBuildValues.PreferBuiltInOrFrameworkType.ToCustomTag()).ToArray());
 
+        protected SimplifyTypeNamesDiagnosticAnalyzerBase()
+            : base(ImmutableDictionary<DiagnosticDescriptor, IOption2>.Empty
+                  .Add(s_descriptorSimplifyNames, null!)
+                  .Add(s_descriptorSimplifyMemberAccess, null!)
+                  .Add(s_descriptorPreferBuiltinOrFrameworkType, null!)) // PROTOTYPE: TODO: This was already broken. We have no ctor that matches what this analyzer has (ie, multiple descriptors, with one descriptor having more than one option).
+        {
+        }
```
Comment (Youssef1313) [inline src/Analyzers/Core/Analyzers/SimplifyTypeNames/SimplifyTypeNamesDiagnosticAnalyzerBase.cs:79]: @mavasani Two of these descriptors don't have options, and one of them has two options, `PreferIntrinsicPredefinedTypeKeywordInMemberAccess` and `PreferIntrinsicPredefinedTypeKeywordInDeclaration`

Are you okay with introducing a new constructor that accepts `ImmutableDictionary<DiagnosticDescriptor, ImmutableHashSet<IOption2>>`?
Or do you have another suggestion?

Comment (Youssef1313) [reply src/Analyzers/Core/Analyzers/SimplifyTypeNames/SimplifyTypeNamesDiagnosticAnalyzerBase.cs]: Actually the options are already added by another analyzer that uses the same ID:

https://github.com/dotnet/roslyn/blob/02aa5ba2d76195a2eba9dd3c90051cc8971201c0/src/Features/Core/Portable/PreferFrameworkType/PreferFrameworkTypeDiagnosticAnalyzerBase.cs#L23

Comment (Youssef1313) [reply src/Analyzers/Core/Analyzers/SimplifyTypeNames/SimplifyTypeNamesDiagnosticAnalyzerBase.cs]: @mavasani This is something to consider as a follow-up. Should we remove `s_descriptorPreferBuiltinOrFrameworkType` from `SimplifyTypeNamesDiagnosticAnalyzerBase` in favor of `PreferFrameworkTypeDiagnosticAnalyzerBase`? They both use the same code-style options and the same diagnostic id.

Comment (Youssef1313) [reply src/Analyzers/Core/Analyzers/SimplifyTypeNames/SimplifyTypeNamesDiagnosticAnalyzerBase.cs]: Tracked by https://github.com/dotnet/roslyn/issues/50307. I forgot I opened such an issue :)

**Comment by Youssef1313** [comment]
@mavasani Can you review please? Thanks!

File: src/Analyzers/Core/Analyzers/SimplifyTypeNames/SimplifyTypeNamesDiagnosticAnalyzerBase.cs (hunk 1 above)
Comment (mavasani) [reply src/Analyzers/Core/Analyzers/SimplifyTypeNames/SimplifyTypeNamesDiagnosticAnalyzerBase.cs]: Everything you mentioned in the above comments sounds good to me!

**Review by mavasani — APPROVED** [review]

**Pull request merged**